File: src/main/java/com/hst/sitescrapper/handler/GlobalExceptionHandler.java
package com.hst.sitescrapper.handler;

import com.hst.sitescrapper.exception.ServiceException;
import com.hst.sitescrapper.model.api.ApiResponse;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(ServiceException.class)
    public ResponseEntity<ApiResponse> handleServiceException(ServiceException e) {
        return new ResponseEntity<>(ApiResponse.of(HttpStatus.INTERNAL_SERVER_ERROR.value(), e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
    }

}

File: backend/src/main/kotlin/com/hst/scrapper/global/handler/GlobalExceptionHandler.kt
package com.hst.scrapper.global.handler

import com.hst.scrapper.global.exception.BaseException
import com.hst.scrapper.global.model.api.ApiResponse
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

/**
 * @author dlgusrb0808@gmail.com
 */
@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(BaseException::class)
    fun handleBaseException(exception: BaseException): ResponseEntity<Map<String, Any>> {
        exception.attribute["message"] = exception.message
        return ResponseEntity<Map<String, Any>>(exception.attribute, exception.status)
    }

}